sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala

```diff
@@ -1857,7 +1857,7 @@ object GpuOverrides {
           }
         }
 
-        override def convertToGpu(child: Expression): GpuExpression = GpuSum(child)
+        override def convertToGpu(child: Expression): GpuExpression = GpuSum(child, a.dataType)
```

Comment:
this kind of change will be needed for avg for decimal support, the current tests are passing. Easiest way to identify expressions requiring this treatment in Spark:
```
➜  spark git:(master) ag 'def dataType: DataType = resultType'
sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/Sum.scala
47:  override def dataType: DataType = resultType

sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/Average.scala
51:  override def dataType: DataType = resultType
```